src/python/pants/backend/jvm/tasks/coverage/manager.py

```diff
@@ -0,0 +1,107 @@
+# coding=utf-8
+# Copyright 2014 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import os
+import shutil
+
+from pants.backend.jvm.tasks.coverage.cobertura import Cobertura
+from pants.backend.jvm.tasks.coverage.engine import NoCoverage
+from pants.backend.jvm.tasks.coverage.jacoco import Jacoco
+from pants.subsystem.subsystem import Subsystem
+from pants.subsystem.subsystem_client_mixin import SubsystemClientMixin
+from pants.util.dirutil import safe_mkdir
+from pants.util.strutil import safe_shlex_split
+
+
+class CodeCoverageSettings(object):
+  """A class containing settings for code coverage tasks."""
+
+  def __init__(self, options, context, workdir, tool_classpath, confs, log,
+               copy2=shutil.copy2, copytree=shutil.copytree, is_file=os.path.isfile,
+               safe_md=safe_mkdir):
+    self.options = options
+    self.context = context
+    self.workdir = workdir
+    self.tool_classpath = tool_classpath
+    self.confs = confs
+    self.log = log
+
+    self.coverage_dir = os.path.join(self.workdir, 'coverage')
+    self.coverage_instrument_dir = os.path.join(self.coverage_dir, 'classes')
+    self.coverage_console_file = os.path.join(self.coverage_dir, 'coverage.txt')
+    self.coverage_xml_file = os.path.join(self.coverage_dir, 'coverage.xml')
+    self.coverage_html_file = os.path.join(self.coverage_dir, 'html', 'index.html')
+
+    self.coverage_jvm_options = []
+    for jvm_option in options.coverage_jvm_options:
+      self.coverage_jvm_options.extend(safe_shlex_split(jvm_option))
+
+    self.coverage_open = options.coverage_open
+    self.coverage_force = options.coverage_force
+
+    # Injecting these methods to make unit testing cleaner.
+    self.copy2 = copy2
+    self.copytree = copytree
+    self.is_file = is_file
+    self.safe_makedir = safe_md
+
+  @classmethod
+  def from_task(cls, task, workdir=None):
+    return cls(options=task.get_options(),
+               context=task.context,
+               workdir=workdir or task.workdir,
+               tool_classpath=task.tool_classpath,
+               confs=task.confs,
+               log=task.context.log)
+
+
+class CodeCoverage(Subsystem, SubsystemClientMixin):
```

Comment:
Not sure the `SubsystemClientMixin` inheritance is necessary, but could be wrong.